precompiles/parachain-staking/src/lib.rs

```diff
@@ -196,7 +196,84 @@ where
 		Ok(selected_candidates)
 	}
 
+	#[precompile::public("delegationAmount(address,address)")]
+	#[precompile::view]
+	fn delegation_amount(
+		handle: &mut impl PrecompileHandle,
+		delegator: Address,
+		candidate: Address,
+	) -> EvmResult<U256> {
+		// Fetch info.
+		handle.record_cost(RuntimeHelper::<Runtime>::db_read_gas_cost())?;
+		let (candidate, delegator) = (
+			Runtime::AddressMapping::into_account_id(candidate.0),
+			Runtime::AddressMapping::into_account_id(delegator.0),
+		);
+		let amount: U256 = if let Some(state) =
+			pallet_parachain_staking::Pallet::<Runtime>::delegator_state(&delegator)
+		{
+			// get the delegation amount
+			if let Some(pallet_parachain_staking::Bond { amount, .. }) = state
+				.delegations
+				.0
+				.into_iter()
+				.find(|b| b.owner == candidate)
+			{
+				amount.into()
+			} else {
+				log::trace!(
+					target: "staking-precompile",
+					"Delegation for {:?} not found, so delegation amount is 0",
+					candidate
+				);
+				U256::zero()
+			}
+		} else {
+			log::trace!(
+				target: "staking-precompile",
+				"Delegator state for {:?} not found, so delegation amount is 0",
+				delegator
+			);
+			U256::zero()
+		};
+
+		Ok(amount)
+	}
+
 	// Role Verifiers
+	#[precompile::public("isInTopDelegations(address,address)")]
+	#[precompile::view]
+	fn is_in_top_delegations(
+		handle: &mut impl PrecompileHandle,
+		delegator: Address,
+		candidate: Address,
+	) -> EvmResult<bool> {
+		let (candidate, delegator) = (
+			Runtime::AddressMapping::into_account_id(candidate.0),
+			Runtime::AddressMapping::into_account_id(delegator.0),
+		);
+
+		// Fetch info.
+		handle.record_cost(RuntimeHelper::<Runtime>::db_read_gas_cost())?;
+		let is_in_top_delegations = if let Some(delegations) =
+			pallet_parachain_staking::Pallet::<Runtime>::top_delegations(&candidate)
+		{
+			delegations
+				.delegations
+				.into_iter()
+				.any(|b| b.owner == delegator)
+		} else {
+			log::trace!(
+				target: "staking-precompile",
+				"Candidate state for {:?} not found, so delegation is not in top",
+				candidate
+			);
+			false
+		};
```

Comment:
Same as above, trace logs are unnecessary here for arbitrary user input.

```suggestion
		let is_in_top_delegations = pallet_parachain_staking::Pallet::<Runtime>::top_delegations(
			&candidate,
		)
		.map_or(false, |delegations| {
			delegations
				.delegations
				.into_iter()
				.any(|b| b.owner == delegator)
		});
```